src/modules/identity/RadixIdentity.ts

```diff
@@ -23,18 +23,18 @@
 import RadixSignatureProvider from './RadixSignatureProvider'
 import RadixDecryptionProvider from './RadixDecryptionProvider'
 import RadixAccount from '../account/RadixAccount'
-import { RadixAddress, RadixAtom } from '../atommodel';
+import { RadixAddress, RadixAtom } from '../atommodel'
 
-export default abstract class RadixIdentity implements RadixSignatureProvider, RadixDecryptionProvider {
-    constructor(readonly address: RadixAddress) {
-        this.account = new RadixAccount(address)
-        this.account.enableDecryption(this)
-    }
-
-    public abstract signAtom(atom: RadixAtom): Promise<RadixAtom>
-    public abstract decryptECIESPayload(payload: Buffer): Promise<Buffer>
-    public abstract decryptECIESPayloadWithProtectors(protectors: Buffer[], payload: Buffer): Promise<Buffer>
-    public abstract getPublicKey(): Buffer
-
-    public account: RadixAccount
+export default interface RadixIdentity extends RadixSignatureProvider, RadixDecryptionProvider {
+    signAtom(atom: RadixAtom): Promise<RadixAtom>
+    decryptECIESPayload(payload: Buffer): Promise<Buffer>
+    decryptECIESPayloadWithProtectors(protectors: Buffer[], payload: Buffer): Promise<Buffer>
+    getPublicKey(): Buffer
+    account: RadixAccount
 }
+
+export function setupIdentityAccount(this, address: RadixAddress): RadixAccount {
```

Comment:
Why this change from a constructor to a function, and from a class to an interface? It's odd, since you need this function in every type of identity, that's what classes are for

Reply:
Ok you win... but I like functions :cry:

Reply:
Functions are great, but I think in this case it's kind of awkward to have to know that you need to import this alongside the interface